server/tserver/src/main/java/org/apache/accumulo/tserver/tablet/Tablet.java

```diff
@@ -586,6 +591,11 @@ private LookupResult lookup(SortedKeyValueIterator<Key,Value> mmfi, List<Range>
           mmfi.next();
         }
 
+        if (yield.hasYielded()) {
+          Key yieldPosition = yield.getPositionAndReset();
```

Comment:
We should probably ensure that an iterator is not returning the same yield key. Suggest ensuring that the yieldPosition is within the scan range.